packages/@aws-cdk/aws-cloudfront/lib/edge-function.ts

```diff
@@ -0,0 +1,209 @@
+import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as iam from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as ssm from '@aws-cdk/aws-ssm';
+import {
+  App, BootstraplessSynthesizer, Construct, ConstructNode, DefaultStackSynthesizer, IStackSynthesizer,
+  ResourceEnvironment, Stack, StackProps, Token,
+} from '@aws-cdk/core';
+import * as resources from '@aws-cdk/custom-resources';
+
+/** Properties for creating a Lambda@Edge function */
+export interface EdgeFunctionProps extends lambda.FunctionProps { }
+
+/**
+ * A Lambda@Edge function.
+ *
+ * Convenience construct for requesting a Lambda function in the 'us-east-1' region for use with Lambda@Edge.
+ * Implements several restrictions enforced by Lambda@Edge.
+ */
+export class EdgeFunction extends Construct implements lambda.IVersion {
```

Comment:
A major limitation to consider with this approach is that customers will not be able to use [NodejsFunction](https://github.com/aws/aws-cdk/blob/e349004a522e2123c1e93bd3402dd7c3f9c5c17c/packages/%40aws-cdk/aws-lambda-nodejs/lib/function.ts#L56), PythonFunction and friends with this construct. 

We should look at how we can enable this.

One naïve option that comes to mind - change `NodejsFunction` and friends to not extend `Function` but create either a `Function` or `EdgeFunction` child construct. This switch can be based on input properties or based on whether the call goes to `edgeArn()` or not.

Reply:
cc @jogold

Reply:
Great call-out.

In your mind, would that make a strong argument for moving this out into its own module? I think it would be a bit weird for `aws-lambda-nodejs` to take a dependency on `aws-cloudfront`.

Reply:
> I think it would be a bit weird for aws-lambda-nodejs to take a dependency on aws-cloudfront

Agreed.